File: BookExamples/33_ViewModelDemo_LiveData/app/src/main/java/com/ebookfrenzy/viewmodeldemo/ui/main/MainViewModel.java
package com.ebookfrenzy.viewmodeldemo.ui.main;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {

    private static final Float usd_to_eu_rate = 0.74F;
    private String dollarText = "";

    /*LINE NUMBER 13 WAS CHANGED FROM
    private Float result = 0F;
     */
    private MutableLiveData<Float> result = new MutableLiveData<>();


    public void setAmount(String value) {
        this.dollarText = value;
        /*
        LINE 23 WAS CHANGED FROM THE CODE BELOW BECAUSE WE HAD TO USE THE setValue() METHOD OF
        THE MUTABLELIVEDATA INSTANCE
        result = Float.valueOf(dollarText) * usd_to_eu_rate;
        */
        result.setValue(Float.valueOf(dollarText)*usd_to_eu_rate);
    }

    /*
    LINE 30 WAS CHANGED FROM THE CODE BELOW BECAUSE IT HAD TO BE PART OF THE MUTABLELIVEDATA
    INSTANCE
    public Float getResult()
     */
    public MutableLiveData<Float> getResult()
    {
        return result;
    }

}

File: class_book_examples/pre_eel/chapter40_ViewModelDemo/app/src/main/java/com/ebookfrenzy/viewmodeldemo/ui/main/MainViewModel.kt
package com.ebookfrenzy.viewmodeldemo.ui.main

import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {
    private val rate = 0.74f
    private var dollarText = ""
    private var result: Float = 0f

    fun setAmount(value: String) {
        this.dollarText = value
        result = value.toFloat() * rate
    }

    fun getResult(): Float {
        return result
    }
}